tests/python/frontend/pytorch/test_forward.py

```diff
@@ -836,6 +836,20 @@ def forward(self, *args):
     input_data = torch.rand(input_shape).float()
     verify_model(Size1().float().eval(), input_data=input_data)
 
+
+def test_type_as():
+    torch.set_grad_enabled(False)
+    input_shape = [1, 3]
+
+    class TypeAsInt32(Module):
+        def forward(self, *args):
+            int32_tensor = torch.zeros(1, 3, dtype=torch.int32)
+            return args[0].type_as(int32_tensor)
```

Comment:
Great stuff!

I would suggest to test more dtypes.
(One of my main goals with #5779 is to be able to use input_types safely, and I hit troubles with it before.)

Reply:
Done. Did not realize the torch types and tvm type has different naming. Updated the test with more types covered.